# Alumno: Daniel Rodriguez Amezaga

## Resumen: intenté realizar estas tareas de todas las formas que pude, me funciona con Dask

##  Modin me falla si uso el csv de la práctica, intente instalar rapids pero no logré hacerlo.

## Creo entender que modin separa la tarea en subtareas más pequeñas aprovechando mejor el prcesamiendo de los datos para poder cargar DataFrames de gran tamaño, carge un dataframe más pequeño y tardo unos 24 segundos pude ver como lo segmentaba en tareas.

## Intente instalar Rapdis con Docker pero no conseguí lenvatar el contenedor usar jupyter y probarlo, de verdad que lo intenté y seguiré haciendolo.

## Características del PC a probar

### Disco duro:
        Tamaño:466G  Usados:203G  
        Disponibles263G  Uso:44% 

### Memory Device
        Total Width: 64 bits
        Data Width: 64 bits
        Size: 4 GB
        Form Factor: SODIMM
        Type: DDR4
        Type Detail: Synchronous
        Speed: 2133 MT/s
        Configured Memory Speed: 2133 MT/s

### Microprocesador
        Architecture: x86_64
        CPU op-mode(s): 32-bit, 64-bit
        Model name: Intel(R) Core(TM) i5-6200U CPU @ 2.30GHz
        CPU MHz: 2295.696
        CPU max MHz: 2800,0000
        CPU min MHz: 400,0000
        CPU(s): 4
        Thread(s) per core:  2
        Core(s) per socket:  2
        Socket(s): 1
        CPU family: 6

# Pandas 

In [1]:
%%time
import pandas as pd

CPU times: total: 375 ms
Wall time: 1.45 s


In [2]:
%%time
# Mi ordenadoer no puede, se bloquea.
# The kernel appears to have died. It will restart automatically.
# Cree un dataframe más pequeño para poder avanzar.
# Usando Dask me funciona todo.
df_pandas = pd.read_csv("dataframes/yellow_tripdata_2018-01-prueba.csv")
df_pandas.head()

CPU times: total: 1.69 s
Wall time: 1.67 s


VendorID tpep_pickup_datetime tpep_dropoff_datetime  passenger_count  \
0       NaN                  NaN                   NaN              NaN   
1       1.0  2018-01-01 00:21:05   2018-01-01 00:24:23              1.0   
2       1.0  2018-01-01 00:44:55   2018-01-01 01:03:05              1.0   
3       1.0  2018-01-01 00:08:26   2018-01-01 00:14:21              2.0   
4       1.0  2018-01-01 00:20:22   2018-01-01 00:52:51              1.0   

   trip_distance  RatecodeID store_and_fwd_flag  PULocationID  DOLocationID  \
0            NaN         NaN                NaN           NaN           NaN   
1            0.5         1.0                  N          41.0          24.0   
2            2.7         1.0                  N         239.0         140.0   
3            0.8         1.0                  N         262.0         141.0   
4           10.2         1.0                  N         140.0         257.0   

   payment_type  fare_amount  extra  mta_tax  tip_amount  tolls_amount  \
0           NaN          NaN    NaN      NaN         NaN           NaN   
1           2.0          4.5    0.5      0.5         0.0           0.0   
2           2.0         14.0    0.5      0.5         0.0           0.0   
3           1.0          6.0    0.5      0.5         1.0           0.0   
4           2.0         33.5    0.5      0.5         0.0           0.0   

   improvement_surcharge  total_amount  
0                    NaN           NaN  
1                    0.3           5.8  
2                    0.3          15.3  
3                    0.3           8.3  
4                    0.3          34.8

In [3]:
%%time
df_pandas.describe()

CPU times: total: 953 ms
Wall time: 927 ms


VendorID  passenger_count  trip_distance    RatecodeID  \
count  1.043127e+06     1.043127e+06   1.043127e+06  1.043127e+06   
mean   1.571752e+00     1.713981e+00   3.038615e+00  1.042834e+00   
std    4.948251e-01     1.267974e+00   3.385668e+00  4.185707e-01   
min    1.000000e+00     0.000000e+00   0.000000e+00  1.000000e+00   
25%    1.000000e+00     1.000000e+00   1.000000e+00  1.000000e+00   
50%    2.000000e+00     1.000000e+00   1.900000e+00  1.000000e+00   
75%    2.000000e+00     2.000000e+00   3.730000e+00  1.000000e+00   
max    2.000000e+00     9.000000e+00   1.178700e+02  9.900000e+01   

       PULocationID  DOLocationID  payment_type   fare_amount         extra  \
count  1.043127e+06  1.043127e+06  1.043127e+06  1.043127e+06  1.043127e+06   
mean   1.570947e+02  1.552785e+02  1.397498e+00  1.287874e+01  4.157750e-01   
std    7.094488e+01  7.479942e+01  5.162298e-01  1.109005e+01  1.907559e-01   
min    1.000000e+00  1.000000e+00  1.000000e+00 -1.980000e+02 -1.000000e+00   
25%    1.000000e+02  9.000000e+01  1.000000e+00  6.500000e+00  5.000000e-01   
50%    1.590000e+02  1.580000e+02  1.000000e+00  9.500000e+00  5.000000e-01   
75%    2.320000e+02  2.330000e+02  2.000000e+00  1.600000e+01  5.000000e-01   
max    2.650000e+02  2.650000e+02  4.000000e+00  5.888500e+02  4.800000e+00   

            mta_tax    tip_amount  tolls_amount  improvement_surcharge  \
count  1.043127e+06  1.043127e+06  1.043127e+06           1.043127e+06   
mean   4.963945e-01  1.681446e+00  1.953465e-01           2.995151e-01   
std    4.571449e-02  2.618763e+00  1.324790e+00           1.600583e-02   
min   -5.000000e-01 -1.000000e+00 -1.050000e+01          -3.000000e-01   
25%    5.000000e-01  0.000000e+00  0.000000e+00           3.000000e-01   
50%    5.000000e-01  1.100000e+00  0.000000e+00           3.000000e-01   
75%    5.000000e-01  2.360000e+00  0.000000e+00           3.000000e-01   
max    5.000000e-01  1.917000e+02  5.675600e+02           3.000000e-01   

       total_amount  
count  1.043127e+06  
mean   1.597889e+01  
std    1.301733e+01  
min   -2.093000e+02  
25%    8.300000e+00  
50%    1.230000e+01  
75%    1.896000e+01  
max    6.203600e+02

In [4]:
%%time
df_pandas.fare_amount.value_counts()

CPU times: total: 15.6 ms
Wall time: 19 ms


5.50      49860
6.00      49600
5.00      48830
6.50      46708
7.00      42725
          ...  
400.00        1
142.50        1
37.56         1
14.30         1
98.50         1
Name: fare_amount, Length: 450, dtype: int64

In [5]:
%%time
len(df_pandas), df_pandas.shape

CPU times: total: 0 ns
Wall time: 0 ns


(1043128, (1043128, 17))

In [6]:
%%time
df_pandas.tail()

CPU times: total: 0 ns
Wall time: 0 ns


VendorID tpep_pickup_datetime tpep_dropoff_datetime  passenger_count  \
1043123       1.0  2018-01-01 02:28:35   2018-01-01 02:46:06              1.0   
1043124       1.0  2018-01-01 02:49:28   2018-01-01 03:17:05              1.0   
1043125       2.0  2018-01-01 02:10:14   2018-01-01 02:34:54              1.0   
1043126       2.0  2018-01-01 00:58:00   2018-01-01 01:13:40              2.0   
1043127       2.0  2018-01-01 01:14:37   2018-01-01 01:25:20              2.0   

         trip_distance  RatecodeID store_and_fwd_flag  PULocationID  \
1043123           2.50         1.0                  N         261.0   
1043124           5.90         1.0                  N         148.0   
1043125          13.51         1.0                  N         229.0   
1043126           3.79         1.0                  N          88.0   
1043127           1.84         1.0                  N          68.0   

         DOLocationID  payment_type  fare_amount  extra  mta_tax  tip_amount  \
1043123         148.0           2.0         13.0    0.5      0.5         0.0   
1043124         188.0           2.0         23.0    0.5      0.5         0.0   
1043125         173.0           2.0         38.0    0.5      0.5         0.0   
1043126          68.0           2.0         14.5    0.5      0.5         0.0   
1043127         230.0           2.0          9.0    0.5      0.5         0.0   

         tolls_amount  improvement_surcharge  total_amount  
1043123          0.00                    0.3         14.30  
1043124          0.00                    0.3         24.30  
1043125          5.76                    0.3         45.06  
1043126          0.00                    0.3         15.80  
1043127          0.00                    0.3         10.30

# DASK 

In [7]:
%%time
import dask.dataframe as dd

CPU times: total: 422 ms
Wall time: 542 ms


In [8]:
%%time
df_dask = dd.read_csv("dataframes/yellow_tripdata_2018-01.csv", 
                     assume_missing=True)
df_dask.head()

CPU times: total: 1.36 s
Wall time: 1.41 s


VendorID tpep_pickup_datetime tpep_dropoff_datetime  passenger_count  \
0       1.0  2018-01-01 00:21:05   2018-01-01 00:24:23              1.0   
1       1.0  2018-01-01 00:44:55   2018-01-01 01:03:05              1.0   
2       1.0  2018-01-01 00:08:26   2018-01-01 00:14:21              2.0   
3       1.0  2018-01-01 00:20:22   2018-01-01 00:52:51              1.0   
4       1.0  2018-01-01 00:09:18   2018-01-01 00:27:06              2.0   

   trip_distance  RatecodeID store_and_fwd_flag  PULocationID  DOLocationID  \
0            0.5         1.0                  N          41.0          24.0   
1            2.7         1.0                  N         239.0         140.0   
2            0.8         1.0                  N         262.0         141.0   
3           10.2         1.0                  N         140.0         257.0   
4            2.5         1.0                  N         246.0         239.0   

   payment_type  fare_amount  extra  mta_tax  tip_amount  tolls_amount  \
0           2.0          4.5    0.5      0.5        0.00           0.0   
1           2.0         14.0    0.5      0.5        0.00           0.0   
2           1.0          6.0    0.5      0.5        1.00           0.0   
3           2.0         33.5    0.5      0.5        0.00           0.0   
4           1.0         12.5    0.5      0.5        2.75           0.0   

   improvement_surcharge  total_amount  
0                    0.3          5.80  
1                    0.3         15.30  
2                    0.3          8.30  
3                    0.3         34.80  
4                    0.3         16.55

In [9]:
%%time
df_dask.describe()

CPU times: total: 93.8 ms
Wall time: 93.8 ms


Dask DataFrame Structure:
              VendorID passenger_count trip_distance RatecodeID PULocationID DOLocationID payment_type fare_amount    extra  mta_tax tip_amount tolls_amount improvement_surcharge total_amount
npartitions=1                                                                                                                                                                                  
               float64         float64       float64    float64      float64      float64      float64     float64  float64  float64    float64      float64               float64      float64
                   ...             ...           ...        ...          ...          ...          ...         ...      ...      ...        ...          ...                   ...          ...
Dask Name: describe-numeric, 63 graph layers

In [10]:
%%time
df_dask.describe().compute

CPU times: total: 93.8 ms
Wall time: 92.8 ms


<bound method DaskMethodsMixin.compute of Dask DataFrame Structure:
              VendorID passenger_count trip_distance RatecodeID PULocationID DOLocationID payment_type fare_amount    extra  mta_tax tip_amount tolls_amount improvement_surcharge total_amount
npartitions=1                                                                                                                                                                                  
               float64         float64       float64    float64      float64      float64      float64     float64  float64  float64    float64      float64               float64      float64
                   ...             ...           ...        ...          ...          ...          ...         ...      ...      ...        ...          ...                   ...          ...
Dask Name: describe-numeric, 63 graph layers>

In [11]:
%%time
df_dask.fare_amount.value_counts()

CPU times: total: 0 ns
Wall time: 993 µs


Dask Series Structure:
npartitions=1
    int64
      ...
Name: fare_amount, dtype: int64
Dask Name: value-counts-agg, 4 graph layers

In [12]:
%%time
df_dask.fare_amount.value_counts().compute()

CPU times: total: 9.8 s
Wall time: 2.48 s


6.00       473234
5.50       465176
6.50       461927
7.00       446376
5.00       433249
            ...  
60.06           1
60.30           1
60.53           1
60.55           1
8016.00         1
Name: fare_amount, Length: 1714, dtype: int64

# VAEX desde un CSV (1º forma) 

In [13]:
%%time
import vaex

CPU times: total: 859 ms
Wall time: 2.16 s


In [15]:
%%time
# Error: The kernel appears to have died. It will restart automatically.
# Vuelvo a utilizar el mismo CSV reducido
# Al ser mas pequeño me funciona
df_vaex = vaex.from_csv("dataframes/yellow_tripdata_2018-01-prueba.csv", 
                        convert=True)
df_vaex

FileExistsError: [WinError 183] No se puede crear un archivo que ya existe: 'dataframes/yellow_tripdata_2018-01-prueba.csv_chunk_0.hdf5' -> 'dataframes/yellow_tripdata_2018-01-prueba.csv.hdf5'

# VAEX desde HDF5

In [4]:
%%time
import vaex

CPU times: user 12 µs, sys: 0 ns, total: 12 µs
Wall time: 19.6 µs


In [16]:
%%time
vaex_hdf5 = vaex.open("yellow_tripdata_2018-01-prueba.csv.hdf5")
vaex_hdf5

CPU times: user 36.5 ms, sys: 11.5 ms, total: 48 ms
Wall time: 84.4 ms


#        VendorID    tpep_pickup_datetime    tpep_dropoff_datetime    passenger_count    trip_distance    RatecodeID    store_and_fwd_flag    PULocationID    DOLocationID    payment_type    fare_amount    extra    mta_tax    tip_amount    tolls_amount    improvement_surcharge    total_amount
0        nan         --                      --                       nan                nan              nan           --                    nan             nan             nan             nan            nan      nan        nan           nan             nan                      nan
1        1.0         2018-01-01 00:21:05     2018-01-01 00:24:23      1.0                0.5              1.0           N                     41.0            24.0            2.0             4.5            0.5      0.5        0.0           0.0             0.3                      5.8
2        1.0         2018-01-01 00:44:55     2018-01-01 01:03:05      1.0                2.7              1.0           N                     239.0           140.0           2.0             14.0           0.5      0.5        0.0           0.0             0.3                      15.3
3        1.0         2018-01-01 00:08:26     2018-01-01 00:14:21      2.0                0.8              1.0           N                     262.0           141.0           1.0             6.0            0.5      0.5        1.0           0.0             0.3                      8.3
4        1.0         2018-01-01 00:20:22     2018-01-01 00:52:51      1.0                10.2             1.0           N                     140.0           257.0           2.0             33.5           0.5      0.5        0.0           0.0             0.3                      34.8
...      ...         ...                     ...                      ...                ...              ...           ...                   ...             ...             ...             ...            ...      ...        ...           ...             ...                      ...
286,299  1.0         2018-01-02 09:21:15     2018-01-02 09:30:54      1.0                2.8              1.0           N                     246.0           231.0           2.0             11.0           0.0      0.5        0.0           0.0             0.3                      11.8
286,300  2.0         2018-01-02 09:13:49     2018-01-02 09:34:12      1.0                4.56             1.0           N                     230.0           13.0            1.0             18.0           0.0      0.5        3.76          0.0             0.3                      22.56
286,301  2.0         2018-01-02 09:52:14     2018-01-02 10:11:30      1.0                2.78             1.0           N                     114.0           162.0           1.0             14.0           0.0      0.5        2.96          0.0             0.3                      17.76
286,302  2.0         2018-01-02 09:03:47     2018-01-02 09:20:47      5.0                5.21             1.0           N                     75.0            244.0           1.0             18.0           0.0      0.5        3.76          0.0             0.3                      22.56
286,303  1.0         2018                    --                       nan                nan              nan           --                    nan             nan             nan             nan            nan      nan        nan           nan             nan                      nan

In [17]:
%%time
vaex_hdf5.describe()

CPU times: user 919 ms, sys: 61.7 ms, total: 981 ms
Wall time: 923 ms


VendorID tpep_pickup_datetime tpep_dropoff_datetime  \
data_type            float64               string                string   
count                 286303               286303                286302   
NA                         1                    1                     2   
mean       1.590598072671261                   --                    --   
std                 0.491723                   --                    --   
min                      1.0                   --                    --   
max                      2.0                   --                    --   

              passenger_count       trip_distance          RatecodeID  \
data_type             float64             float64             float64   
count                  286302              286302              286302   
NA                          2                   2                   2   
mean       1.6917101522168898  3.3421989018588736  1.0635901949689488   
std                  1.288109            4.219785            0.551496   
min                       0.0                 0.0                 1.0   
max                       9.0              117.87                99.0   

          store_and_fwd_flag        PULocationID        DOLocationID  \
data_type             string             float64             float64   
count                 286302              286302              286302   
NA                         2                   2                   2   
mean                      --  159.29911771486053  156.93577760546555   
std                       --           68.219472           72.460117   
min                       --                 1.0                 1.0   
max                       --               265.0               265.0   

                 payment_type         fare_amount                extra  \
data_type             float64             float64              float64   
count                  286302              286302               286302   
NA                          2                   2                    2   
mean       1.4135178936926742  13.101507184721035  0.20913091770228637   
std                  0.516334           12.779077             0.253875   
min                       1.0              -198.0                 -1.0   
max                       4.0              588.85                  4.8   

                       mta_tax         tip_amount         tolls_amount  \
data_type              float64            float64              float64   
count                   286302             286302               286302   
NA                           2                  2                    2   
mean       0.49577020069716593  1.662272809830232  0.38551092901898854   
std                   0.049301           2.711947              1.97216   
min                       -0.5               -1.0                -10.5   
max                        0.5              191.7               567.56   

          improvement_surcharge        total_amount  
data_type               float64             float64  
count                    286302              286302  
NA                            2                   2  
mean        0.29947502986317087  16.160452913370424  
std                     0.01669           15.460551  
min                        -0.3              -209.3  
max                         0.3              620.36

In [18]:
%%time
vaex_hdf5.fare_amount.value_counts()

CPU times: user 21.1 ms, sys: 3.54 ms, total: 24.7 ms
Wall time: 38.2 ms


 5.50      16395
 5.00      15815
 6.00      15580
 6.50      14848
 4.50      13777
           ...  
 3.20          1
 142.50        1
 68.76         1
-20.00         1
 179.50        1
Length: 451, dtype: int64

# VAEX (2º forma) 

In [5]:
%%time
import vaex

CPU times: user 9 µs, sys: 1 µs, total: 10 µs
Wall time: 17.6 µs


In [20]:
%%time
df_vaex_3 = vaex.from_csv("yellow_tripdata_2018-01-prueba.csv")
df_vaex_3

CPU times: user 1.28 s, sys: 222 ms, total: 1.5 s
Wall time: 1.54 s


#        VendorID    tpep_pickup_datetime    tpep_dropoff_datetime    passenger_count    trip_distance    RatecodeID    store_and_fwd_flag    PULocationID    DOLocationID    payment_type    fare_amount    extra    mta_tax    tip_amount    tolls_amount    improvement_surcharge    total_amount
0        nan         --                      --                       nan                nan              nan           --                    nan             nan             nan             nan            nan      nan        nan           nan             nan                      nan
1        1.0         2018-01-01 00:21:05     2018-01-01 00:24:23      1.0                0.5              1.0           N                     41.0            24.0            2.0             4.5            0.5      0.5        0.0           0.0             0.3                      5.8
2        1.0         2018-01-01 00:44:55     2018-01-01 01:03:05      1.0                2.7              1.0           N                     239.0           140.0           2.0             14.0           0.5      0.5        0.0           0.0             0.3                      15.3
3        1.0         2018-01-01 00:08:26     2018-01-01 00:14:21      2.0                0.8              1.0           N                     262.0           141.0           1.0             6.0            0.5      0.5        1.0           0.0             0.3                      8.3
4        1.0         2018-01-01 00:20:22     2018-01-01 00:52:51      1.0                10.2             1.0           N                     140.0           257.0           2.0             33.5           0.5      0.5        0.0           0.0             0.3                      34.8
...      ...         ...                     ...                      ...                ...              ...           ...                   ...             ...             ...             ...            ...      ...        ...           ...             ...                      ...
286,299  1.0         2018-01-02 09:21:15     2018-01-02 09:30:54      1.0                2.8              1.0           N                     246.0           231.0           2.0             11.0           0.0      0.5        0.0           0.0             0.3                      11.8
286,300  2.0         2018-01-02 09:13:49     2018-01-02 09:34:12      1.0                4.56             1.0           N                     230.0           13.0            1.0             18.0           0.0      0.5        3.76          0.0             0.3                      22.56
286,301  2.0         2018-01-02 09:52:14     2018-01-02 10:11:30      1.0                2.78             1.0           N                     114.0           162.0           1.0             14.0           0.0      0.5        2.96          0.0             0.3                      17.76
286,302  2.0         2018-01-02 09:03:47     2018-01-02 09:20:47      5.0                5.21             1.0           N                     75.0            244.0           1.0             18.0           0.0      0.5        3.76          0.0             0.3                      22.56
286,303  1.0         2018                    --                       nan                nan              nan           --                    nan             nan             nan             nan            nan      nan        nan           nan             nan                      nan

In [21]:
%%time
df_vaex_3.export("yellow_tripdata_2018-01-prueba.csv_exportado.hdf5")

CPU times: user 635 ms, sys: 171 ms, total: 806 ms
Wall time: 822 ms


In [22]:
%%time
df_vaex_4 = vaex.open("yellow_tripdata_2018-01-prueba.csv_exportado.hdf5")
df_vaex_4

CPU times: user 42.6 ms, sys: 0 ns, total: 42.6 ms
Wall time: 56 ms


#        VendorID    tpep_pickup_datetime    tpep_dropoff_datetime    passenger_count    trip_distance    RatecodeID    store_and_fwd_flag    PULocationID    DOLocationID    payment_type    fare_amount    extra    mta_tax    tip_amount    tolls_amount    improvement_surcharge    total_amount
0        nan         --                      --                       nan                nan              nan           --                    nan             nan             nan             nan            nan      nan        nan           nan             nan                      nan
1        1.0         2018-01-01 00:21:05     2018-01-01 00:24:23      1.0                0.5              1.0           N                     41.0            24.0            2.0             4.5            0.5      0.5        0.0           0.0             0.3                      5.8
2        1.0         2018-01-01 00:44:55     2018-01-01 01:03:05      1.0                2.7              1.0           N                     239.0           140.0           2.0             14.0           0.5      0.5        0.0           0.0             0.3                      15.3
3        1.0         2018-01-01 00:08:26     2018-01-01 00:14:21      2.0                0.8              1.0           N                     262.0           141.0           1.0             6.0            0.5      0.5        1.0           0.0             0.3                      8.3
4        1.0         2018-01-01 00:20:22     2018-01-01 00:52:51      1.0                10.2             1.0           N                     140.0           257.0           2.0             33.5           0.5      0.5        0.0           0.0             0.3                      34.8
...      ...         ...                     ...                      ...                ...              ...           ...                   ...             ...             ...             ...            ...      ...        ...           ...             ...                      ...
286,299  1.0         2018-01-02 09:21:15     2018-01-02 09:30:54      1.0                2.8              1.0           N                     246.0           231.0           2.0             11.0           0.0      0.5        0.0           0.0             0.3                      11.8
286,300  2.0         2018-01-02 09:13:49     2018-01-02 09:34:12      1.0                4.56             1.0           N                     230.0           13.0            1.0             18.0           0.0      0.5        3.76          0.0             0.3                      22.56
286,301  2.0         2018-01-02 09:52:14     2018-01-02 10:11:30      1.0                2.78             1.0           N                     114.0           162.0           1.0             14.0           0.0      0.5        2.96          0.0             0.3                      17.76
286,302  2.0         2018-01-02 09:03:47     2018-01-02 09:20:47      5.0                5.21             1.0           N                     75.0            244.0           1.0             18.0           0.0      0.5        3.76          0.0             0.3                      22.56
286,303  1.0         2018                    --                       nan                nan              nan           --                    nan             nan             nan             nan            nan      nan        nan           nan             nan                      nan

#  Usando Modin

In [4]:
%%time
import modin.pandas as pd

CPU times: user 182 ms, sys: 13.1 ms, total: 195 ms
Wall time: 633 ms


In [ ]:
%%time
df = pd.read_csv("yellow_tripdata_2018-01.csv")
df.head()

(raylet) [2023-02-26 18:36:25,353 E 25302 25302] (raylet) node_manager.cc:3097: 2 Workers (tasks / actors) killed due to memory pressure (OOM), 0 Workers crashed due to other reasons at node (ID: 1f89f4d13601f4acebd0d31395080ba1a2f224e72503d6d93ca577ae, IP: 192.168.18.3) over the last time period. To see more information about the Workers killed on this node, use `ray logs raylet.out -ip 192.168.18.3`
(raylet) 
(raylet) Refer to the documentation on how to address the out of memory issue: https://docs.ray.io/en/latest/ray-core/scheduling/ray-oom-prevention.html. Consider provisioning more memory on this node or reducing task parallelism by requesting more CPUs per task. To adjust the kill threshold, set the environment variable `RAY_memory_usage_threshold` when starting Ray. To disable worker killing, set the environment variable `RAY_memory_monitor_refresh_ms` to zero.
(raylet) Error processing line 1 of /usr/local/lib/python3.9/dist-packages/modin-autoimport-pandas.pth:
(raylet) 
(ra

### Mi equipo no puede
(raylet)       import six
(raylet)   ModuleNotFoundError: No module named 'six'
(raylet) 
(raylet) Remainder of file ignored
(raylet) [2023-02-26 19:10:27,423 E 25302 25302] (raylet) node_manager.cc:3097: 3 Workers (tasks / actors) killed due to memory pressure (OOM), 0 Workers crashed due to other reasons at node (ID: 1f89f4d13601f4acebd0d31395080ba1a2f224e72503d6d93ca577ae, IP: 192.168.18.3) over the last time period. To see more information about the Workers killed on this node, use `ray logs raylet.out -ip 192.168.18.3`
(raylet) 
(raylet) Refer to the documentation on how to address the out of memory issue: https://docs.ray.io/en/latest/ray-core/scheduling/ray-oom-prevention.html. Consider provisioning more memory on this node or reducing task parallelism by requesting more CPUs per task. To adjust the kill threshold, set the environment variable `RAY_memory_usage_threshold` when starting Ray. To disable worker killing, set the environment variable `RAY_memory_monitor_refresh_ms` to zero.

# Cuarta parte de la actividad
### A elegir entre Rapids ó Koalas (aunque pudiera ser otro distinto a los mencionados si se prefiere). Revisar en internet su documentación, y elegir uno de ellos, del cual se va a hacer algo (lo que se quiera) en base a lo que el/la estudiante aprenda.

#### Rapdis:
##### https://docs.rapids.ai/install#docker

Configuro con docker y me sale este codigo para instalar:

docker pull nvcr.io/nvidia/rapidsai/rapidsai-core:23.02-cuda11.8-runtime-ubuntu22.04-py3.10

### la terminal se queda parada y no responde
docker run --gpus all --rm -it --shm-size=1g --ulimit memlock=-1 --ulimit stack=67108864 -p 8888:8888 -p 8787:8787 -p 8786:8786 nvcr.io/nvidia/rapidsai/rapidsai-core:23.02-cuda11.8-runtime-ubuntu22.04-py3.10